server/repository/src/db_diesel/document.rs

```diff
@@ -0,0 +1,322 @@
+use super::StorageConnection;
+
+use crate::{db_diesel::json_schema::json_schema, diesel_macros::apply_equal_filter};
+use crate::{EqualFilter, RepositoryError};
+
+use chrono::{DateTime, NaiveDateTime, Utc};
+use diesel::prelude::*;
+
+table! {
+    document (id) {
+        id -> Text,
+        name -> Text,
+        parents -> Text,
+        author -> Text,
+        timestamp -> Timestamp,
+        #[sql_name = "type"] type_ -> Text,
+        data -> Text,
+        schema_id -> Nullable<Text>,
+    }
+}
+
+joinable!(document -> json_schema (schema_id));
+
+allow_tables_to_appear_in_same_query!(document, json_schema);
+
+#[derive(Clone, Queryable, Insertable, AsChangeset, Debug, PartialEq)]
+#[table_name = "document"]
+pub struct DocumentRow {
+    /// The document data hash
+    pub id: String,
+    /// Document path and name
+    pub name: String,
+    /// Stringified array of parents
+    pub parents: String,
+    /// Id of the author who edited this document version
+    pub author: String,
+    /// The timestamp of this document version
+    pub timestamp: NaiveDateTime,
+    /// Type of the containing data
+    #[column_name = "type_"]
+    pub r#type: String,
+    /// The actual document data
+    pub data: String,
+    /// JSON schema id containing the schema for the data
+    pub schema_id: Option<String>,
+}
+
+table! {
```

Comment:
as per our convo i think we don't need document_head, it just means we have to maintain the table state quite in different places (i suppose it could be useful if we store the whole data, but if it's just changes, then it shouldn't be needed)

Reply:
We can put it back in as a local aux table when performance becomes an issue